### ISSUE TITLE
[Bug] Session id / operation id are not isolated by share level.

### ISSUE THREAD

**Issue by HaoYang670** [apache/kyuubi#5256]
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the bug

In USER level and GROUP level anyone can access to other users' / groups' Session and Operation as long as they have the session id or operation id. This breaks the share level isolation and may cause resource leak.

### Affects Version(s)

master

### Kyuubi Server Configurations

```yaml
Disable authentication and authorization.
```

### Are you willing to submit PR?

- [ ] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to fix.
- [X] No. I cannot submit a PR at this time.

**Comment by yaooqinn** [comment]
Perhaps it would be beneficial to include some redaction on these crucial handles.

**Comment by HaoYang670** [comment]
What do you mean for redaction?

**Comment by yaooqinn** [comment]
https://kyuubi.readthedocs.io/en/v1.7.0/search.html?q=redaction, we have redactions for sensitive information in configurations. I guess we can apply something similar to these handles

**Comment by HaoYang670** [comment]
Sorry, it could be a workaround but not the final solution. I guess what Kyuubi should do is to check the access permission for sessions and operations, otherwise Kyuubi engines can't be isolated.

**Comment by yaooqinn** [comment]
How can you `Disable authentication and authorization.` and check the access permission? Sorry, I am confused

**Comment by HaoYang670** [comment]
authentication and share level isolation are 2 different things, right?

**Comment by HaoYang670** [comment]
user1 can't get access to user2's stuff (on USER share level) even if the authN is disabled, am I wrong?

**Comment by HaoYang670** [comment]
I verified for both authN enabled and  disabled and the problem is always there

**Comment by yaooqinn** [comment]
OK, I got your point. One solution might be to reuse the secret field of THandleIdentifier

**Comment by HaoYang670** [comment]
Is it easy to fix

**Comment by yaooqinn** [comment]
yes, it's just partially revert https://github.com/apache/kyuubi/pull/2829/files
